### PR TITLE
CTCP-1012 Adding v2 support

### PR DESCRIPTION
**Pull request by ardwalker**
This is the first part of the ticket, adding support for V2 endpoints

### CODE REVIEW

File: app/v2/controllers/stream/StreamingParsers.scala [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import play.api.libs.streams.Accumulator
+import play.api.mvc.BaseControllerHelpers
+import play.api.mvc.BodyParser
+import play.api.mvc.Request
+
+import java.util.concurrent.Executors
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+trait StreamingParsers {
```
Comment (drnaylor-oc) [inline app/v2/controllers/stream/StreamingParsers.scala:35]: Quite a lot of this trait is unnecessary for this project - we only care about `streamFromMemory` here as that's all we'll use, I suggest removing the other options from here.

**Review by drnaylor-oc — APPROVED** [review]
Nothing particularly wrong here, my comment is just an observation really that could be addressed in your next PR as I don't want to block you further, but it obviates the need for some common respository, whether that be a library or a submodule -- the duplication of code across projects is getting a little silly now (which is not your fault!)

File: test/controllers/DepartureTestMessagesControllerSpec.scala [hunk 2]
```diff
@@ -72,7 +72,8 @@ class DepartureTestMessagesControllerSpec
     with IntegrationPatience
     with TestXml {
 
-  val departureId = new DepartureId(1)
+  val departureString = "1"
+  val departureId     = new DepartureId(departureString.toInt)
```
Comment (derekspalmer) [inline test/controllers/DepartureTestMessagesControllerSpec.scala:76]: case Class

File: test/controllers/DepartureTestMessagesControllerSpec.scala [hunk 3]
```diff
@@ -72,7 +72,8 @@ class DepartureTestMessagesControllerSpec
     with IntegrationPatience
     with TestXml {
 
-  val departureId = new DepartureId(1)
+  val departureString = "1"
```
Comment (derekspalmer) [inline test/controllers/DepartureTestMessagesControllerSpec.scala:75]: Do we need the String as part of the name?

File: test/routing/VersionedRoutingSpec.scala [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+
+  class Harness(cc: ControllerComponents)(implicit val materializer: Materializer)
+      extends BackendController(cc)
+      with VersionedRouting
+      with StreamingParsers
+      with Logging {
+
+    def testWithContent: Action[Source[ByteString, _]] = route {
+      case Some("application/vnd.hmrc.2.0+json") => contentActionTwo
+      case Some(x) if x != MimeTypes.TEXT        => contentActionOne
```
Comment (derekspalmer) [inline test/routing/VersionedRoutingSpec.scala:69]: case  _  ?

**Pull request merged**